docs/content/patterns/crud-data-tables/index.mdx

```diff
@@ -0,0 +1,448 @@
+
+Provide an ‘Edit’ link in the last column of the table so users can open and update or edit existing records. The edit link can be included in a drop down of actions in the last column if there are more than two actions required.
+
+Editing content in place in semantic tables is difficult for some users, so use a side page or side drawer to provide all inputs and controls required to edit the record. When a user elects to edit a record open the side drawer or side page and show the populated fields to be edited.
+
+After the record is edited in the side page or drawer, return the user to the table page, show the edited record in the table, and provide a Section alert immediately above the table to show success. Focus the Section alert so the change is announced to screen reader users.
+
+The update or edit pattern is repeatable and scalable.
+
+<ImageWithBorder
+	alt="Screenshot of an open side drawer with the title ‘Add record’ and a populated short form with text input, radio buttons, a Select component and form buttons ‘Add record’ and ‘Cancel’"
+	src="/img/patterns/crud-tables-edit-record.png"
+/>
+
+### Delete a record from a data table
+
+Provide a ’Delete’ link in the last column of the table so users can delete existing records. The delete link can be included in a drop down of actions in the last column if there are more than two actions required.
+
+If the table supports batch actions more than one record can be deleted at the same time.
+
+Before any record is deleted provide a destructive action modal to make sure a user intends to remove the record.
+
+After the record is deleted, remove the record from the table, and provide a Section alert immediately above the table to show success. Focus the Section alert so the change is announced to screen reader users.
+
+The delete pattern is repeatable and scalable.
+
+<ImageWithBorder
+	alt="Screenshot of an open Modal dialog with the title ‘Are you sure you want to remove this record’ and description ‘It will be removed immediately. You can't undo this action.’"
+	src="/img/patterns/crud-tables-delete-record.png"
+/>
+
+## Upload an undefined list of files to a table
+
+Use a repeatable file upload pattern when uploading an undefined list of files of different categories or when specific metadata for each file is required. Metadata can be associated with uploaded files when using this pattern to support back stage processing.
+
+The upload undefined list of files pattern reduces the amount of form content on the parent page making it easier for users to complete the form, and keeps the users context and focus when providing all required metadata in the side drawer or subpage.
+
+### Empty state
+
+Before any files are uploaded provide an empty state. Use a standard empty state rather than an empty table because the experience of the empty state will be better for screen reader users. They will not need to navigate through a table to find out it is empty.
```

Comment:
```suggestion
Before any files are uploaded, provide an empty state. Use a standard empty state rather than an empty table because the experience of the empty state is simpler for many users. Screen reader users will not need to navigate through a table to find out it is empty.
```